guppylang-internals/src/guppylang_internals/decorator.py

```diff
@@ -201,6 +209,14 @@ def dec(c: type[T]) -> type[T]:
 def wasm_module(
     filename: str,
 ) -> Callable[[builtins.type[T]], GuppyDefinition]:
+    wasm_file = pathlib.Path(filename)
+    if wasm_file.is_file():
+        with pathlib.Path.open(wasm_file, "rb") as f:
+            wasm_bytes = f.read()
+            wasm_sigs = decode_wasm_functions(filename, wasm_bytes)
+    else:
+        raise GuppyError(WasmFileNotFound(None, filename))
```

Comment:
I think this would be better as a regular Python exception since it's invoked when the decorator is called, not when the program is compiled